moveit_planners/ompl/ompl_interface/include/moveit/ompl_interface/detail/ompl_constraints.h

```diff
@@ -0,0 +1,293 @@
+};
+
+/** \brief Position and orientation constraint
+ *
+ * Combine position constraints modelled as a box with orientation constraints
+ * using exponential coordinates, but ignore the last rotation around the z-axis.
+ * This is a common constraint for grasping, welding, ...
+ * */
+class PoseConstraint : public BaseConstraint
+{
+public:
+  PoseConstraint(robot_model::RobotModelConstPtr robot_model, const std::string& group, const unsigned int num_dofs)
+    : BaseConstraint(robot_model, group, num_dofs, 5)
+  {
+  }
+
+  void jacobian(const Eigen::Ref<const Eigen::VectorXd>& x, Eigen::Ref<Eigen::MatrixXd> out) const override
+  {
+    ob::Constraint::jacobian(x, out);
+  }
+
+  virtual void parseConstraintMsg(const moveit_msgs::Constraints& constraints) override;
+  virtual Eigen::VectorXd calcError(const Eigen::Ref<const Eigen::VectorXd>& x) const override;
+  virtual Eigen::MatrixXd calcErrorJacobian(const Eigen::Ref<const Eigen::VectorXd>& x) const override;
+};
+
+/** \brief Extract position constraints from the MoveIt message.
+ *
+ * Assumes there is a single primitive of type box.
+ * Only the dimensions of this box are used here.
+ * These are bounds on the deviation of the end-effector from
+ * the desired position given as the position of the box in the field
+ * constraint_regions.primitive_poses[0].position.
+ *
+ * @todo: also use the link name in future?
+ * Now we assume the constraints are for the end-effector link.
+ * */
+std::vector<Bounds> positionConstraintMsgToBoundVector(const moveit_msgs::PositionConstraint& pos_con);
+
+/** \brief Extract orientation constraints from the MoveIt message
+ *
+ * These bounds are assumed to be centered around the nominal orientation / desired orientation
+ * given in the field "orientation" in the message.
+ * These bounds are therefore bounds on the orientation error between the desired orientation
+ * and the current orientation of the end-effector.
+ *
+ * The three bounds x, y, and z, can be applied to different parameterizations of the rotation error.
+ * (Roll, pithc, and yaw or exponential coordinates or something else.)
+ *
+ * */
+std::vector<Bounds> orientationConstraintMsgToBoundVector(const moveit_msgs::OrientationConstraint& ori_con);
+
+/** \brief Factory to create constraints based on what is in the MoveIt constraint message. **/
+std::shared_ptr<BaseConstraint> createConstraint(robot_model::RobotModelConstPtr robot_model, const std::string& group,
+                                                 const moveit_msgs::Constraints& constraints);
+
+/** \brief Conversion matrix to go from angular velocity in the world frame to
+ * angle axis equivalent.
+ *
+ * Based on:
+ * https://ethz.ch/content/dam/ethz/special-interest/mavt/robotics-n-intelligent-systems/rsl-dam/documents/RobotDynamics2016/RD2016script.pdf
+ *
+ * */
+Eigen::Matrix3d angularVelocityToAngleAxis(double angle, const Eigen::Vector3d& axis);
```

Comment:
I should change the name, as it is not converting to Angle-Axis representation, but to exponential coordinates. Although the difference is really minimal.

Reply:
Yes, I would change the name. The output is not exactly the same as what the name suggest.